src/tike/ptycho/ptycho.py

```diff
@@ -234,142 +263,162 @@ def reconstruct(
+    """Return `n` boolean masks that split the field of view into stripes.
+
+    Mask divide the data into spatially contiguous regions along the position
+    axis.
+
+    Split scan into three stripes:
+    >>> [scan[:, s] for s in split_by_scan_stripes(scan, 3)]
+
+    FIXME: Only uses the first view to divide the positions. Assumes the
     positions on all angles are distributed similarly.
 
     Parameters
     ----------
-    pool : ThreadPool
-    scan (ntheta, nscan, 2) float32
-        Scan positions.
-    data (ntheta, nscan // fly, D, D) float32
-        Captured frames from the detector.
+    scan : (ntheta, nscan, 2) float32
+        The 2D coordinates of the scan positions.
+    n : int
+        The number of stripes.
     fly : int
-        The number of scan positions per data frame
+        The number of scan positions per frame.
+    axis : int (0 or 1)
+        Which spatial dimension to divide along. i.e. horizontal or vertical.
 
     Returns
     -------
-    scan_list, data_list : list, list
-        Scan positions and data split into chunks and scattered to GPUs
+    mask : list of (nscan, ) boolean
+        A list of boolean arrays which divide the scan positions into `n`
+        stripes.
+
     """
-    # Reshape scan so positions in the same fly scan are not separated
+    if scan.ndim != 3:
+        raise ValueError('scan must have three dimensions.')
+    if n < 1:
+        raise ValueError('The number of stripes must be > 0.')
+
     ntheta, nscan, _ = scan.shape
+    if (nscan // fly) * fly != nscan:
+        raise ValueError('The number of scan positions must be an '
+                         'integer multiple of the number of fly positions.')
+
+    # Reshape scan so positions in the same fly scan are not separated
     scan = scan.reshape(ntheta, nscan // fly, fly, 2)
+
     # Determine the edges of the horizontal stripes
     edges = np.linspace(
-        0,
-        scan[..., 0].max(),
-        pool.num_workers + 1,
+        scan[..., axis].min(),
+        scan[..., axis].max(),
+        n + 1,
         endpoint=True,
     )
-    # Split the scan positions and data amongst the stripes
-    scan_list, data_list = [], []
-    for i in range(pool.num_workers):
-        keep = np.logical_and(
-            edges[i] < scan[0, :, 0, 0],
-            scan[0, :, 0, 0] <= edges[i + 1],
-        )
-        scan_list.append(scan[:, keep].reshape(ntheta, -1, 2).astype('float32'))
-        data_list.append(data[:, keep].astype('float32'))
-    scan_list = list(pool.scatter(scan_list))
-    data_list = list(pool.scatter(data_list))
-    return scan_list, data_list
+
+    # Move the outer edges to include all points
+    edges[0] -= 1
+    edges[-1] += 1
+
+    # Generate masks which put points into stripes
+    return [
+        np.logical_and(
+            edges[i] < scan[0, :, 0, axis],
+            scan[0, :, 0, axis] <= edges[i + 1],
+        ).repeat(fly) for i in range(n)
+    ]
+
+
+if __name__ == "__main__":
```

Comment:
Is this for testing?

Reply:
Yes! I forgot to move these to the test module. Good catch.